controllers/designate_controller.go

```diff
@@ -1118,14 +1157,88 @@ func (r *DesignateReconciler) generateServiceConfigMaps(
 		},
 	}
 
-	err := oko_secret.EnsureSecrets(ctx, h, instance, cms, envVars)
+	err = oko_secret.EnsureSecrets(ctx, h, instance, cms, envVars)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
+func (r *DesignateReconciler) createAndStoreRndcKeySecrets(
+	ctx context.Context,
+	h *helper.Helper,
+	instance *designatev1beta1.Designate,
+	cmLabels map[string]string,
+	numOfSecretsToCreate int,
+) error {
+	for i := 0; i < numOfSecretsToCreate; i++ {
```

Comment:
you always start at 0, so if 3 secrets exist (secret-0, secret-1, secret-2) and you need an additional one, it creates secret-0 intead of secret-3?

Reply:
Yes, I missed that, thank you. I should use the replica count instead of the delta.